src/compiler/cell_manager.rs

```diff
@@ -163,8 +197,27 @@ pub struct MaxWidthCellManager {
 
 impl CellManager for MaxWidthCellManager {
     fn place<F, StepArgs>(&self, unit: &mut CompilationUnit<F, StepArgs>) {
+        // If there exists shared signals, determine whether MaxWidthCellManager will return steps
```

Comment:
So for this CM specifically I think it is better to panic always if there are shared columns. We could revisit this in the future

Reply:
Fixed. Removed associated code for shared columns and panic at the top if there are shared columns.